pkg/extkingpin/path_content_reloader_test.go

```diff
@@ -98,6 +102,10 @@ func TestPathContentReloader(t *testing.T) {
 			testutil.Ok(t, err)
 
 			tt.args.runSteps(t, testFile, pathContent)
+			if tt.wantReloads == 0 {
+				// Give things a little time to sync. The fs watcher events are heavily async and could be delayed.
+				time.Sleep(1 * time.Second)
```

Comment:
Crossposting from https://github.com/thanos-io/thanos/commit/c6623f80fa7305b573947987d14ded5cd3bcbd32#r120681785. Please check `testutil.Equals(t, tt.wantReloads, reloadCount)` in a loop instead or something. This `time.Sleep` will probably lead to flakiness.

Reply:
@GiedriusS I have some other follow ups to this part of the code and will include this change for the next PR. 👍

Reply:
But this loop will have a timeout though, as I don't want a test failure to end up with an infinite loop or rely on `test.timeout` when running tests.

Reply:
👍 you can use runutil.Repeat with a context to achieve that

Reply:
@GiedriusS I actually have to keep the sleep there for one reason: a real scenario might have a delayed event from the fs and the test might pass if the assertion runs quickly before the event arrives... but would fail, as expected, if it had waited slightly more. 

I'm adding the repeat there though for all the other scenarios.